core/arch/arm/plat-imx/registers/imx6.h

```diff
@@ -140,4 +140,31 @@
 #define DIGPROG_OFFSET	0x260
 #endif
 
+#if defined(CFG_MX6ULL)
+#define I2C_CLK_RATE		66000000
```

Comment:
if the BSP configured the peripheral clock to run of the 24MHz oscillator, the definition above is unnecessary.

Reply:
I added a comment to the imx6.h file; since the upstream uboot uses the 66MHz clock instead of the 24MHz oscillator.

Reply:
Can't you tell the clock from DT?

Reply:
yes...I need to add proper dt support; I send a patch upstream (uboot) to default to the 24MHz oscillator...let me remove this support for 66MHz so we can get it merge and then will work out the dt configuration as we add support for more boards. hope that is acceptable

Reply:
Sure, DT support can be added later. In the mean time I would recommend using a config variable in `plat-imx/conf.mk`, for instance `CFG_IMX_i2C_CLK_RATE ?= 24000000` (or whichever value is more convenient as a default...).

Reply:
ok